helix-view/src/clipboard.rs

```diff
@@ -20,6 +25,33 @@ macro_rules! command_provider {
                 prg: $set_prg,
                 args: &[ $( $set_arg ),* ],
             },
+            get_primary_cmd: None,
+            set_primary_cmd: None,
+            selection_buf: String::new(),
+        })
+    }};
+}
+
+macro_rules! command_provider_with_primary {
+    (paste => $get_prg:literal $( , $get_arg:literal )* ; copy => $set_prg:literal $( , $set_arg:literal )* ; primary_paste => $pr_get_prg:literal $( , $pr_get_arg:literal )* ; primary_copy => $pr_set_prg:literal $( , $pr_set_arg:literal )* ; ) => {{
```

Comment:
Could you format this line to fit on screen ? It's currently 250 columns wide (rustfmt won't touch it since it's a macro). Also how about `command_provider_with_primary` simply being another rule in `command_provider` ?